spec/scheduling-tasks.md

```diff
@@ -169,14 +183,36 @@ A <dfn>scheduler task queue</dfn> is a [=struct=] with the following [=struct/it
 </div>
 
 <div algorithm>
-  To <dfn>create a scheduler task queue</dfn> with {{TaskPriority}} |priority|:
+  To <dfn>create a scheduler task queue</dfn> with {{TaskPriority}} |priority| and |removalSteps|:
 
   1. Let |queue| be a new [=scheduler task queue=].
   1. Set |queue|'s [=scheduler task queue/priority=] to |priority|.
   1. Set |queue|'s [=scheduler task queue/tasks=] to a new empty [=set=].
+  1. Set |queue|'s [=scheduler task queue/removal steps=] to |removalSteps|.
```

Comment:
A new task queue starts empty and has removal steps attached.

After the last task is rejected/completed, we will call the removal steps, and those steps I think make this queue no longer valid without re-creating.

I don't think it can happen per the changes in this change, but it seems like a task-queue must never stay empty after creation, nor ever return back from non-empty?

Reply:
I think this comment refers to the risk of having a reference to the TQ but the queue having been removed from the map? That would indeed be a problem -- and I did have to rewrite things to make sure that can't happen (specifically in the delayed task section, which previously got the TQ when postTask called, but didn't actually queue the task until the delay expired).

A task queue could go from empty to non-empty in the same task, e.g. the last task is dequeued, runs, queues more tasks to the queue within that task. But in that case it's okay because it would still be in the map.

Implementations might want to do the cleanup differently, e.g. in Chrome we use a weak map from signal --> task queue and let GC take care of cleanup (similar for abort algorithms -- we have a GCed handle). But this isn't observable since the task queues themselves aren't web-exposed. Adding the cleanup explicitly here at least makes it so the spec isn't leaky, and indicates when resources could be released.